src/SprykerEco/Zed/Computop/Business/Payment/Handler/Saver/Complete/PayPalExpressCompleteResponseSaver.php

```diff
@@ -0,0 +1,183 @@
+        $this->savePaymentComputop($payPalExpressCompleteResponseTransfer, $computopPaymentTransfer);
+        $this->savePaymentComputopOrderItems($payPalExpressCompleteResponseTransfer, $computopPaymentTransfer);
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\ComputopApiPayPalExpressCompleteResponseTransfer $computopApiPayPalExpressCompleteResponseTransfer
+     * @param \Generated\Shared\Transfer\ComputopPaymentComputopTransfer $computopPaymentComputopTransfer
+     *
+     * @return void
+     */
+    protected function savePaymentComputop(
+        ComputopApiPayPalExpressCompleteResponseTransfer $computopApiPayPalExpressCompleteResponseTransfer,
+        ComputopPaymentComputopTransfer $computopPaymentComputopTransfer
+    ): void {
+        $computopPaymentComputopTransfer->setPayId($computopApiPayPalExpressCompleteResponseTransfer->getHeader()->getPayId());
+        $computopPaymentComputopTransfer->setXId($computopApiPayPalExpressCompleteResponseTransfer->getHeader()->getXId());
+
+        $this->computopEntityManager->saveComputopPayment($computopPaymentComputopTransfer);
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\ComputopApiPayPalExpressCompleteResponseTransfer $payPalExpressCompleteResponseTransfer
+     * @param \Generated\Shared\Transfer\ComputopPaymentComputopTransfer $computopPaymentComputopTransfer
+     *
+     * @return void
+     */
+    protected function savePaymentComputopDetail(
+        ComputopApiPayPalExpressCompleteResponseTransfer $payPalExpressCompleteResponseTransfer,
+        ComputopPaymentComputopTransfer $computopPaymentComputopTransfer
+    ): void {
+        $computopPaymentComputopDetailTransfer = $this
+            ->computopRepository
+            ->getComputopPaymentDetail($computopPaymentComputopTransfer);
+
+        $computopPaymentComputopDetailTransfer->fromArray($payPalExpressCompleteResponseTransfer->toArray(), true);
+
+        $this->computopEntityManager->updateComputopPaymentDetail($computopPaymentComputopDetailTransfer);
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\ComputopApiPayPalExpressCompleteResponseTransfer $completeResponseTransfer
+     * @param \Generated\Shared\Transfer\ComputopPaymentComputopTransfer $computopPaymentComputopTransfer
+     *
+     * @return void
+     */
+    protected function savePaymentComputopOrderItems(
+        ComputopApiPayPalExpressCompleteResponseTransfer $completeResponseTransfer,
+        ComputopPaymentComputopTransfer $computopPaymentComputopTransfer
+    ): void {
+        $salesOrderItemsCollection = $this->computopRepository
+            ->getComputopSalesOrderItemsCollection($computopPaymentComputopTransfer);
+
+        $computopPaymentComputopOrderItemsCollection = $this->computopRepository
+            ->getComputopPaymentComputopOrderItemsCollection($computopPaymentComputopTransfer);
+
+        foreach ($salesOrderItemsCollection->getComputopSalesOrderItems() as $computopSalesOrderItemTransfer) {
```

Comment:
is it possible to save this in bulk? At least via Propel's functionality? (in this case we will have less open transactions)